### ISSUE TITLE
Fields to specify camera position... and maybe more?

### ISSUE THREAD

**Issue by thijsdhollander** [MRtrix3/mrtrix3#287]
So as part of #286 , it came up that there is for instance no way to specify the camera position numerically in mrview at the moment; while mrview does allow specific numerical input for almost everything else I can think of.  Currently, the closest way for a user to "hack" themselves into such a possibility, is to abuse the screen capture tool.  So here's the issue for that.

Maybe (just generally suggesting here) this is a good opportunity to take a step back and also "review" the view options tool again a bit.  I'm not putting out any strong arguments here, but from a certain point of view you could say it's sometimes a bit of a weird mashup of certain options.  Take for instance the focus position: is that really a "view option"?  It is in a certain sense, since it allows slice selection, but then again that's just a consequence of where the focus position happens to be.  Another random thing could be the whole issue/debate on all the lighting settings.  Should they be unified across tools?  If so, are the view options a good place to put them in?  Does this make the view tool so overcrowded that another (sub)tool is required?  Like a camera tool?  Or an appearance tool?

And most of all: does whatever opinion on these things justify the time?  Especially since we might be keen on getting a release out... and so on and so on...

So, without making strong statements already: this issue might be a good place to collect some opinions! :-)

**Comment by jdtournier** [comment]
Just a few thoughts:

- I have no issues with the focus being in the view options - it certainly affects things in orthoview. But it's also probably the most sensible place to find this, unless you have a separate 'navigation' tool to set the volume currently displayed along with the focus...? But then you could argue the projection (axial, coronal, sagittal) probably ought to be in there too. It gets messy, and in my opinion doesn't necessarily help the user find the actual feature they need. Personally I'm happy with it where it is now.

- a single set of lighting parameters isn't in my opinion a good idea. I find the best settings for the ODFs are quite different from those for the streamlines. The one setting that would make sense to unify is the light position. But that would mean having a 'material properties' panel for each tool, and a common 'light properties' panel. I think that'll probably end up being more confusing... That said, I don't tend to change the light position all that much, so maybe that would be OK - we could even put it in the view panel...

More generally, I think it would be good to 'promote' the view tool so it's no longer a humble tool, but a much more visible and obvious part of the main UI. We've talked about this before, and I don't think we'd come up with a decent idea about what that might look like. Maybe it's time to revisit the idea...?

**Comment by thijsdhollander** [comment]
Yep, I definitely wouldn't create extra tools and spread options/features around if there's no good reason in general.  The only reason for rethinking this might be if -for some reason- the view tool would get *too* crowded.  But I also agree that's not a current issue.

Again yes on the second bullet as well.  I'm often tripping up over this one: when I mention a single set of lighting parameters I'm also only thinking of the light itself (currently that means indeed only its position, but that should be enough as well, unless we want to turn things into a disco).

And triple yes: I've always found the view tool, and mostly the fact that it *is* part of the tool list, at least a tad weird as well.  I'm not sure how other people feel about this, but to me "a tool" should be something that provides additional functionality to the basic (main) image viewer.  Most of the current tools are overlay type of things, which nicely fit that feeling of "add on" (they even literally add something "on" your main image).  The screen capture tool is not an overlay, but actually even yet more of an "add on" in its own right: one could go as far as to say that it's not even providing any strict added "viewing" functions to the viewer (but more exporting, or something), so again it makes total sense to be a tool.

...but the view tool provides all sorts of basic functionality that directly relates to interacting with the *main* image.

It does provide *a lot*, so it's probably not a good thing to cram all those things in menus instead.  Maybe it would help if it weren't in the tool list, but could be brought up directly by its own button in the main toolbar?  Another risky additional suggestion (don't hesitate to shoot it down, I'm not even sure myself I like the idea): maybe it should by default pop up as a dock on the *left* side of the window, rather than on the right side like all the tools?  I'm suggesting this since I find myself often needing it in conjunction with another tool.  In many scenarios, using a tool to typically overlay something, I'm not going to (want to) get rid of the main image.  So I might want to tune some appearance/view settings of both.  Now I'm sometimes repeatedly switching back and forth (until this drives me to actually move the view options to dock left, or undock).  Argument against it: in the default mrview window size, having docks left *and* right, means losing almost two thirds of the actual viewing area.  But then again, I'm often maximising the window (almost by default)... so not sure. :confused: 

Final note on suddenly remembering things that can not be set "numerically": the positions and angles of clip planes.  But I'm not sure how this could be provided in a clean manner in the interface... (and of course, if it's needed).

**Comment by Lestropie** [comment]
> Maybe it would help if it weren't in the tool list, but could be brought up directly by its own button in the main toolbar?

This was my thinking... Still behave as a tabbable dockable widget, just accessible from a different location.

**Comment by roeysc** [comment]
Hello everyone,

I was wondering if there is now an option to set the camera position through the command line. I saw some discussion related to this topic (and the fear of confusing between neurological and radiological conventions), suggesting this option was not added to MRView, but perhaps I am missing something.

Also, I will make use of the "and maybe more" in the title and ask: Is it possible to to change the order of the tracks vs image slices? In other words, is it possible for a slice to hide that part of the tracks which is behind it?

Many thanks!
Roey

**Comment by Lestropie** [comment]
Hi Roey,

It doesn't look like anybody has been able to get to this. We've lost a couple of key devs, and have extra overhead trying to organise both the release and the workshop, so haven't had the time to get to non-crucial features. It's still on the list though.

Regarding your second question, can I safely re-phrase this as: "Is it possible to show all track segments in front of a slice, but none of the segments behind it?". If so, you should be able to achieve this with a bit of trickery: Use the volume render view mode, with the alpha slider set to maximum, add a clip plane at the slice of interest, and turn off "crop to slab" in the tractography tool. 

Rob

**Comment by roeysc** [comment]
Hey Rob,
Thanks for the detailed answers!
Regarding the second question, your rephrasing is perfect. I will try this.
Thank you,
Roey
